### PR TITLE
Some changes to CI

### PR DESCRIPTION
**Pull request by giordano**
* run Travis also on macOS and Windows
* run Travis only on master branch
* run Travis on latest stable Julia version
* cache artifacts
* build documentation in a separate job

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,17 +1,44 @@
 language: julia
 os:
   - linux
+  - osx
+  - windows
```
Comment (stevengj) [inline .travis.yml:5]: Not sure what this accomplishes other than to slow down CI … this is a pure-Julia package where portability hasn't been an issue.

Comment (giordano) [reply .travis.yml]: I don't have a strong opinion about them and I removed these lines, even though it happened to me in the past to find platform-dependent Julia quirks in pure-Julia packages

**Comment by stevengj** [comment]
Thanks!   Hopefully nothing new needs be done permissions-wise to make the new documentation builds work.

**Comment by giordano** [comment]
If it was working already before, nothing new should be needed, I only moved documentation to its own job, as suggested in Documenter: https://juliadocs.github.io/Documenter.jl/stable/man/hosting/#Travis-CI

**Pull request merged**